src/battle_util.c

```diff
@@ -9060,7 +9060,10 @@ static u32 CalcFinalDmg(u32 dmg, u16 move, u8 battlerAtk, u8 battlerDef, u8 move
     }
 
     // check stab
-    if (IS_BATTLER_OF_TYPE(battlerAtk, moveType) && move != MOVE_STRUGGLE && move != MOVE_NONE)
+    // Pokemon will get STAB after type change, so include in the AI decision dmg calcs
```

Comment:
This shouldn't go there, as this calculates the actual damage when a move is used. Since it's an AI check, I suggest moving it to AI-only part of the code

Reply:
The AI uses the same damage calculations when deciding on a move to make.
AI_CalcDamage -> CalculateMoveDamageAndEffectiveness   -> DoMoveDamageCalc -> CalcFinalDmg

Protean & Libero need to replace the STAB boost modifier when running AI Calcs, since it will always get the STAB bonus,  but cannot be applied additive earlier in the AI logic or it could potentially get ability boost 1.5x and Stab 1.5x.

I don't think we should break out the AI damage calcs to be seperate from the actual damage calcs either, since this would create code redundancy and future changes to calcs would need to be made in multiple spots.